ledger/src/blockstore_processor.rs

```diff
@@ -949,6 +971,14 @@ pub fn confirm_slot(
 
     timing.execute_timings.accumulate(&execute_timings);
 
+    if !check_result.finish_verify() {
+        warn!("Ledger proof of history failed at slot: {}", slot);
+        return Err(BlockError::InvalidEntryHash.into());
```

Comment:
If this fails, should we kill the poh verification thread?

Reply:
@ryleung-solana did you see this comment?

Reply:
> even before this change, with just the old synchronous CPU sigverify path, we also did not cancel the poh verification if the sigverify failed due to this

This isn't quite true. Before we still waited for poh verification to finish before returning the sigverify error. Is there any risk to any of that memory passed to the gpu thread being longer lived than maybe expected?

Reply:
Have changed the logic in confirm_slot to always wait for all async operations to finish even in the error case.

Reply:
Sorry I was mistaken here, you were right that we already weren't waiting for poh verification to continue.